precompiles/erc20/approve.go

```diff
@@ -183,9 +174,9 @@ func (p Precompile) DecreaseAllowance(
 	case subtractedValue != nil && subtractedValue.Cmp(common.Big0) <= 0:
 		// case 1. subtractedValue 0 or negative -> return error
 		err = errors.New("cannot decrease allowance with non-positive values")
-	case authorization == nil:
+	case err != nil:
 		// case 2. no authorization -> return error
-		err = errors.New("allowance does not exist")
+		err = sdkerrors.Wrapf(err, "allowance does not exist")
```

Comment:
checking for the error here because it returns the specific reason the allowance was not found returned from the `GetAuthzAndAllowance` function